lib/streamlit/server/Server.py

```diff
@@ -506,11 +514,16 @@ def open(self):
         self._session = self._server._create_report_session(self)
 
     def on_close(self):
+        if not self._session:
```

Comment:
@kantuni, in this and the next function, added some early bailouts so`self._session` is guaranteed to be non-`None` in the function body.